src/woo_publications/api/permissions.py

```diff
@@ -17,3 +20,28 @@ def has_permission(self, request: Request, view: APIView) -> bool:
             parameter.name for parameter in ALL_AUDIT_PARAMETERS if parameter.required
         ]
         return all(header in request.headers for header in required_headers)
+
+
+class TokenAuthPermission(BasePermission):
+    def has_permission(self, request, view) -> bool:
+        token = request.auth
+
+        if not isinstance(token, Application):
+            return False
+
+        if not isinstance(token.permissions, list):
+            return False
```

Comment:
this will always be a list, because the model defines it as an `ArrayField`. If this line is here to satisfy the type checker, then do:

```py
assert isinstance(token.permissions, list)
```

now there is missing branch coverage for a situation that cannot happen